internal/upgrade/v0_11_0.go

```diff
@@ -0,0 +1,198 @@
+	if err != nil {
+		// ingester does not exist, maybe scaled down?
+		if client.IgnoreNotFound(err) == nil {
+			return nil
+		}
+		return err
+	}
+
+	patch := ingester.DeepCopy()
+	zero := int32(0)
+	patch.Spec.Replicas = &zero
+	err = u.Client.Patch(ctx, patch, client.MergeFrom(ingester))
+	if err != nil {
+		return err
+	}
+
+	return wait.PollUntilContextTimeout(ctx, pollInterval, pollTimeout, true, func(ctx context.Context) (done bool, err error) {
+		ingester := &appsv1.StatefulSet{}
+		err = u.Client.Get(ctx, ingesterQuery, ingester)
+		if err != nil {
+			return false, err
+		}
+		if ingester.Status.Replicas == 0 {
+			return true, nil
+		}
+
+		return false, nil
+	})
+}
+
+func chown_pvcs(ctx context.Context, u Upgrade, tempo metav1.Object, nodeSelector map[string]string, image string, pvcs *corev1.PersistentVolumeClaimList) error {
+	var volumes []corev1.Volume
+	var volumeMounts []corev1.VolumeMount
+	for _, pvc := range pvcs.Items {
+		volumes = append(volumes, corev1.Volume{
+			Name: pvc.Name,
+			VolumeSource: corev1.VolumeSource{
+				PersistentVolumeClaim: &corev1.PersistentVolumeClaimVolumeSource{
+					ClaimName: pvc.Name,
+				},
+			},
+		})
+
+		volumeMounts = append(volumeMounts, corev1.VolumeMount{
+			Name:      pvc.Name,
+			MountPath: fmt.Sprintf("/var/tempo/%s", pvc.Name),
+		})
+	}
+
+	// keep the jobs around for 1 day
+	ttl := int32(60 * 60 * 24)
+	rootUser := int64(0)
+	upgradeJob := batchv1.Job{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      fmt.Sprintf("chown-%s", tempo.GetName()),
+			Namespace: tempo.GetNamespace(),
+		},
+		Spec: batchv1.JobSpec{
+			Template: corev1.PodTemplateSpec{
+				Spec: corev1.PodSpec{
+					// Make sure the job runs on the same node as ingester
+					NodeSelector:       nodeSelector,
+					ServiceAccountName: naming.DefaultServiceAccountName(tempo.GetName()),
+					Volumes:            volumes,
+					Containers: []corev1.Container{
+						{
+							Name:         "chown",
+							Image:        image,
+							Command:      []string{"chown", "-R", "10001:10001", "/var/tempo"},
+							VolumeMounts: volumeMounts,
+						},
+					},
+					RestartPolicy: corev1.RestartPolicyNever,
+					SecurityContext: &corev1.PodSecurityContext{
+						RunAsUser: &rootUser,
+					},
+				},
+			},
+			TTLSecondsAfterFinished: &ttl,
+		},
+	}
+
+	if err := ctrl.SetControllerReference(tempo, &upgradeJob, u.Client.Scheme()); err != nil {
+		return err
+	}
+	err := u.Client.Create(ctx, &upgradeJob)
+	if err != nil {
+		return err
+	}
+	return wait.PollUntilContextTimeout(ctx, pollInterval, pollTimeout, true, func(ctx context.Context) (done bool, err error) {
+		job := &batchv1.Job{}
+		objectKey := client.ObjectKey{
+			Namespace: upgradeJob.Namespace,
+			Name:      upgradeJob.Name,
+		}
+		err = u.Client.Get(ctx, objectKey, job)
+		if err != nil {
+			return false, err
+		}
+		if job.Status.Succeeded == 1 {
+			return true, nil
+		}
+
+		return false, nil
+	})
+
```

Comment:
we could remove the job here, instead of leaving it there until the 24h TTL. but it's fine either way.

Reply:
I left it there intentionally for awareness :)